cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,639 @@
+        gnd_probs.append(1.0 - np.mean(excited_probs_l))
+
+    return RBResult(numbers_cliffords, gnd_probs)
+
+
+def two_qubit_randomized_benchmarking(sampler: sim.SimulatesSamples,
+                                      first_qubit: devices.GridQubit,
+                                      second_qubit: devices.GridQubit,
+                                      numbers_cliffords: Sequence[int],
+                                      number_circuits: int,
+                                      number_shots: int) -> RBResult:
+    """Clifford-based randomized benchmarking (RB) of two qubits.
+
+    A total of number_circuits random circuits are generated, each of which
+    contains a fixed number of two-qubit Clifford gates plus one additional
+    Clifford that inverts the whole sequence and a measurement in the
+    z-basis. Each circuit is repeated number_shots times and the average
+    |00> state population is determined from the measurement outcomes of all
+    of the circuits.
+
+    The above process is done for different numbers of Cliffords specified in
+    numbers_cliffords.
+
+    The two-qubit Cliffords here are decomposed into CZ gates plus single-qubit
+    x and y rotations. See Barends et al., Nature 508, 500 for details.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        first_qubit: The first qubit under test.
+        second_qubit: The second qubit under test.
+        numbers_cliffords: The different numbers of Cliffords in the RB study.
+        number_circuits: The number of random circuits generated for each
+            number of Cliffords.
+        number_shots: The number of repetitions of each circuit.
+
+    Returns:
+        A RBResult object that stores and plots the result.
+    """
+    c1, _, s1, s1_x, s1_y = _single_qubit_cliffords()
+    cfd_matrices = _two_qubit_clifford_matrices(first_qubit, second_qubit,
+                                                c1, s1, s1_x, s1_y)
+    gnd_probs = []
+    for num_cfds in numbers_cliffords:
+        gnd_probs_l = []
+        for _ in range(number_circuits):
+            circuit = _random_two_q_clifford(first_qubit, second_qubit,
+                                             num_cfds, cfd_matrices,
+                                             c1, s1, s1_x, s1_y)
+            circuit.append(ops.measure(first_qubit, second_qubit, key='z'))
+            results = sampler.run(circuit, repetitions=number_shots)
+            gnds = [(not r[0] and not r[1]) for r in results.measurements['z']]
+            gnd_probs_l.append(np.mean(gnds))
+        gnd_probs.append(float(np.mean(gnd_probs_l)))
+
+    return RBResult(numbers_cliffords, gnd_probs)
+
+
+def single_qubit_state_tomography(sampler: sim.SimulatesSamples,
+                                  qubit: devices.GridQubit,
+                                  circuit: circuits.Circuit,
+                                  number_shots: int) -> TomographyResult:
```

Comment:
I want to see to rename this to `repetitions`, but it feels different from the other experiments. Like the handle the user wants is different here. Unsure.

Reply:
I renamed num_shots to repetitions. Although it does feel a bit different, it still is the parameter that gets fed to the repetitions parameter in the .run method.